### PR TITLE
Implement AnalyserNode

### PR DESCRIPTION
**Pull request by Manishearth**
Do not merge yet, I need to ensure it works on the servo side.

r? @ferjm

### CODE REVIEW

**Comment by Manishearth** [comment]
Okay, this seems to work! r? @ferjm

File: audio/src/analyser_node.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    fn apply_blackman_window(&mut self) {
+        self.compute_blackman_windows();
+        self.windowed.resize(self.fft_size, 0.);
+
+        let mut data_idx = self.convert_index(0);
+        for n in 0..self.fft_size {
+            self.windowed[n] = self.blackman_windows[n] * self.data[data_idx];
+            self.advance_index(&mut data_idx);
+        }
+        println!("windowed {:?}", self.windowed);
```
Comment (ferjm) [inline audio/src/analyser_node.rs]: Remove this line, please

File: audio/src/analyser_node.rs [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    fn compute_fft(&mut self) {
+        if self.fft_computed {
+            return;
+        }
+        self.fft_computed = true;
+        self.apply_blackman_window();
+        self.computed_fft_data.resize(self.fft_size / 2, 0.);
+        self.smoothed_fft_data.resize(self.fft_size / 2, 0.);
+
+        for k in 0..(self.fft_size / 2) {
```
Comment (ferjm) [inline audio/src/analyser_node.rs]: `(self.fft_size / 2) - 1` ?

Comment (Manishearth) [reply audio/src/analyser_node.rs]: Nope, Rust ranges are open on the other end. `0..n` means "zero to n including zero but not n". There's `0..=n` for ranges closed on both ends.

**Pull request merged**